dotnet/src/Functions/Functions.OpenApi.Extensions/Extensions/CopilotAgentPluginOpenApiDocumentVisitor.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Services;
+
+namespace Microsoft.SemanticKernel.Plugins.OpenApi.Extensions;
+
+/// <summary>
+/// Copilot Agent Plugin OpenAPI document visitor that:
+///  * Normalizes the operation IDs by replacing dots with underscores. So that the operation IDs can be used as function names in semantic kernel.
+///  * Truncates the description to the maximum allowed length.
+///  * Removes properties and parameters with the same name.
+/// </summary>
+internal sealed class CopilotAgentPluginOpenApiDocumentVisitor : OpenApiVisitorBase
+{
+    private const int MaximumDescription = 1000;
+
+    public override void Visit(OpenApiOperation operation)
+    {
+        NormalizeOperationId(operation);
+
+        TruncateOperationDescription(operation);
+
+        RemoveDuplicateOperationProperties(operation);
+    }
+
+    private static void NormalizeOperationId(OpenApiOperation operation)
+    {
+        if (operation is null || operation.OperationId is null)
+        {
+            return;
+        }
+        operation.OperationId = operation.OperationId.Replace('.', '_');
+    }
+
+    private static void TruncateOperationDescription(OpenApiOperation operation)
+    {
+        if (operation.Description?.Length > MaximumDescription)
+        {
+            operation.Description = operation.Description.Substring(0, MaximumDescription);
+        }
+    }
```

Comment:
It looks like similar code is present in this PR, I'm wondering if it can be merged together?
https://github.com/microsoft/semantic-kernel/pull/9880/files#diff-b8975b8e1d0490b986826e4e0ff9f94fdcbf9aa4cee01da43c47006d08448f1cR202-R209

Reply:
Either this PR or the other one will be merged but not both.